includes/class-wc-stripe-customer.php

```diff
@@ -185,6 +190,14 @@ public function update_customer( $args = array() ) {
 
 		if ( ! empty( $response->error ) ) {
 			if ( $this->is_no_such_customer_error( $response->error ) ) {
+				// This can happen when switching the main Stripe account or importing users from another site.
+				if ( 0 === $retries ) {
+					// If not already tried, recreate the customer and then update it.
+					$this->delete_id_from_meta();
+					$this->create_customer();
+					return $this->update_customer( $args, $retries++ );
+				}
```

Comment:
@marcinbot this looks like a great solution! Some thoughts and questions on this:

1. At the moment I see `$retries` is used as a boolean flag (either try to create new customer when it's not found or throw). Do you plan to have more than a single retry in the future? If not does it make sense to switch to boolean flag?

2. Does it make sense to delete current customer id from user meta before a new customer is created? Is it better/safer than creating a customer first and then calling `update_id_in_meta`?

3. Three levels of if statements seem to be a good sign for refactoring and extraction some pieces into their own methods. What do you think?

Reply:
Thanks for the changes! Looks good! 👍